### PR TITLE
Fix docstring decorators

### LINKED ISSUE #11159: [BUG] Mleap documentation encompasses params as .. notes::
### Issues Policy acknowledgement

- [X] I have read and agree to submit bug reports in accordance with the [issues policy](https://www.github.com/mlflow/mlflow/blob/master/ISSUE_POLICY.md)

### Where did you encounter this bug?

Local machine

### Willingness to contribute

Yes. I can contribute a fix for this bug independently.

### MLflow version

Mlflow 2.10.2

### System information

- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**:
- **Python version**:
- **yarn version, if running the dev UI**:

### Describe the problem

`mlflow.utils.annotations.deprecated` and `mlflow.utils.annotations.keyword_only` prepend notes/warnings without properly indenting to the docstring. This doesn't break sphinx rendering when there's one decorator, but it does break rendering when there are multiple decorators.

Example (single - shinx renders): 
```
.. note:: adsf

    description

    Args:
        x: asdf
```

Example (double - shinx doesn't render): 
```
..warning:: asdf
    asdf
.. note:: xyz

    description

    Args:
        x: asdf
```

### Tracking information

MLflow mleap docs

### Code to reproduce issue

NA

### Stack trace

NA

### Other info / logs

NA

### What component(s) does this bug affect?

- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/deployments`: MLflow Deployments client APIs, server, and third-party Deployments integrations
- [X] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/recipes`: Recipes, Recipe APIs, Recipe configs, Recipe Templates
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/server-infra`: MLflow Tracking server backend
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

### What interface(s) does this bug affect?

- [ ] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

### What language(s) does this bug affect?

- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

### What integration(s) does this bug affect?

- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

### PR DESCRIPTION
**Pull request by michael-berk**
<details><summary>&#x1F6E0 DevTools &#x1F6E0</summary>
<p>

[Open in GitHub Codespaces](https://codespaces.new/michael-berk/mlflow/pull/11160?quickstart=1)

#### Install mlflow from this PR

```
pip install git+https://github.com/mlflow/mlflow.git@refs/pull/11160/merge
```

#### Checkout with GitHub CLI

```
gh pr checkout 11160
```

</p>
</details>

…oogle

### Related Issues/PRs

Resolve  #11159

### What changes are proposed in this pull request?
* Add indentation for `keyword_only`, `experimental`, and `deprecated` decorators
* Add indentation utils
* Add tests
* Convert mleap to google
### How is this PR tested?

- [x] Existing unit/integration tests
- [ ] New unit/integration tests
- [x] Manual tests

### Does this PR require documentation update?

- [x] No. You can skip the rest of this section.
- [ ] Yes. I've updated:
  - [ ] Examples
  - [ ] API references
  - [ ] Instructions

#### Is this a user-facing change?

- [x] No. You can skip the rest of this section.
- [ ] Yes. Give a description of this change to be included in the release notes for MLflow users.

#### What component(s), interfaces, languages, and integrations does this PR affect?

Components

- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/deployments`: MLflow Deployments client APIs, server, and third-party Deployments integrations
- [x] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/recipes`: Recipes, Recipe APIs, Recipe configs, Recipe Templates
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/server-infra`: MLflow Tracking server backend
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface

- [ ] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language

- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

Integrations

- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

<a name="release-note-category"></a>

#### How should the PR be classified in the release notes? Choose one:

- [x] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

**Comment by michael-berk** [comment]
https://github.com/mlflow/mlflow/pull/11110#issuecomment-1942904573

**Comment by michael-berk** [comment]
@mlflow-automation autoformat

**Comment by michael-berk** [comment]
@mlflow-automation autoformat

File: mlflow/utils/annotations.py [hunk 1]
```diff
@@ -7,6 +8,26 @@
...
+        docstring_str: string with docstring
+
+    Returns:
+        Whitespace corresponding to the indent of a docstring.
+    """
+
+    if not docstring_str or "\n" not in docstring_str:
+        return ""
+
+    return re.match(r"^\s*", docstring_str.split("\n")[-1]).group()
```
Comment (harupy) [inline mlflow/utils/annotations.py:28]:
```suggestion
    return re.match(r"^\s*", docstring_str.rsplit("\n", 1)[-1]).group()
```

can we use rsplit?

File: mlflow/utils/annotations.py [hunk 2]
```diff
@@ -34,8 +55,10 @@ def f(api: C) -> C:
 
 
 def _experimental(api: C, api_type: str) -> C:
+    indent = _get_min_indent_of_docstring(api.__doc__)
     notice = (
-        f"    .. Note:: Experimental: This {api_type} may change or "
+        indent
+        + f".. Note:: Experimental: This {api_type} may change or "
         + "be removed in a future release without warning.\n\n"
```
Comment (harupy) [inline mlflow/utils/annotations.py:62]:
```suggestion
        f".. Note:: Experimental: This {api_type} may change or "
        "be removed in a future release without warning.\n\n"
```

**Review by harupy — APPROVED** [review]
LGTM!

File: mlflow/utils/annotations.py (hunk 2 above)
Comment (michael-berk) [reply mlflow/utils/annotations.py]: Looks like `()` syntax for multiline strings needs the first arg to be a string, not a variable. I opted for `+` over making the `indent` variable an fstring.

**Pull request merged**